utils/firestore.js

```diff
@@ -3539,6 +3539,196 @@ const updateParticipantCorrection = async (participantData) => {
+        const snapshot = await db.collection('participants').where('Connect_ID', '==', connectId).get();
+        if (snapshot.empty) throw { code: 404, message: 'Participant not found.' };
+
+        const participantData = snapshot.docs[0].data();
+        const siteCode = participantData[healthCareProvider];
+        const token = participantData['token'];
+        const specimenArray = await getSpecimenCollections(token, siteCode);
+
+        let incentiveEligible = false;
+
+        const currentPaymentRoundName = currentPaymentRound; // baseline or future payment rounds
+
+        // Add option to handle future payment rounds here later - left paymentRound as a parameter for now
+        if (participantData[paymentRound][currentPaymentRoundName][eligibleForIncentive] === no) { 
+
+            const module1 = (participantData[baselineSurveyStatusModuleBackgroundOverallHealth] === submitted);
+            const module2 = (participantData[baselineSurveyStatusModuleMedReproHealth] === submitted);
+            const module3 = (participantData[baselineSurveyStatusModuleSmokeAlcoholSun] === submitted);
+            const module4 = (participantData[baselineSurveyStatusModuleLiveAndWork] === submitted);
+            const bloodCollected = (participantData[baselineBloodSampleCollected] === yes) || (participantData[collectionDetails]?.[baseline]?.[clinicalSiteBloodCollected] === yes);
+            
+            if (module1 && module2 && module3 && module4) {
+                if(bloodCollected) {
+                    incentiveEligible = true;
+                }    
+                else {
+                    if (specimenArray.length > 0) {
+                        const baselineResearchCollections = specimenArray.filter(collection => collection[biospecimenVisit] === baseline && collection[collectionSetting] === researchCollectionSetting);
+                    
+                        if (baselineResearchCollections.length != 0) {
+                            baselineResearchCollections.forEach(collection => {
+
+                                const researchBloodTubes = [serumSeparatorTube1, serumSeparatorTube2, heparinTube1, edtaTube1, acdTube1, streckTube];
+                                
+                                researchBloodTubes.forEach(tube => {
+                                    if (collection[tube] && collection[tube][reasonTubeNotCollected] && collection[tube][reasonTubeNotCollected] != participantRefusal) {
+                                        incentiveEligible = true;
+                                    }
+                                });
+                            });
+                        }
+                    }
+                }
+            }
+        }
+        return { 
+            isEligibleForIncentive: incentiveEligible, 
+            participantData 
+        };
+    } catch (error) {
+        console.error('Error checking participant for eligible incentive:', error);
+        throw {
```

Comment:
Your non-400 code error handlers use error.toString(), which I believe will result in '[object Object]'. This mismatch should probably be addressed.